packages/bsky/src/services/feed/index.ts

```diff
@@ -216,31 +225,35 @@ export class FeedService {
     depth = 0,
   ): Promise<FeedHydrationState> {
     const { viewer, dids, uris } = refs
-    const [posts, labels, bam] = await Promise.all([
+    const [posts, gates, labels, bam] = await Promise.all([
       this.getPostInfos(Array.from(uris), viewer),
+      this.gatesByPostUri(Array.from(uris)),
       this.services.label.getLabelsForSubjects([...uris, ...dids]),
       this.services.graph.getBlockAndMuteState(
         viewer ? [...dids].map((did) => [viewer, did]) : [],
       ),
     ])
+
     // profileState for labels and bam handled above, profileHydration() shouldn't fetch additional
-    const [profileState, blocks] = await Promise.all([
+    const [profileState, blocks, lists] = await Promise.all([
       this.services.actor.views.profileHydration(
         Array.from(dids),
         { viewer },
         { bam, labels },
       ),
       this.blocksForPosts(posts, bam),
+      this.listsForPostGates(gates, viewer),
```

Comment:
mer seems like a shame to have to hydrate the lists for every post :/

similar to viewer `canReply` state - is this actually needed in feeds?

may be a paul question

Reply:
Spoke with Paul, sounds like we're going to keep this one for now 👍